src/main/kotlin/three/consulting/epoc/service/CustomerService.kt

```diff
@@ -24,5 +24,15 @@ class CustomerService(private val customerRepository: CustomerRepository) {
         }
         else throw UnableToCreateCustomerException()
     }
+
+    fun updateCustomerForId(customerRequest: CustomerDTO): CustomerDTO {
+        if (customerRequest.id != null) {
+            val customer = Customer(customerRequest)
+            return CustomerDTO(customerRepository.save(customer))
+        }
+        else throw UnableToUpdateCustomerException()
+    }
+
 }
 class UnableToCreateCustomerException : RuntimeException("Cannot create a customer with existing id")
+class UnableToUpdateCustomerException : RuntimeException("Missing customer id")
```

Comment:
Still think this could be more informative. We'll introduce a logger at some point so all exceptions should contain a descriptive message. _Missing customer id_ doesn't tell what was trying to be done, but like the exception name itself says _Unable to update a customer with missing id_ or something along those lines would be better.

Reply:
I get it. So it would be preferable to repeat the the failed operation in the error message, not just give explanation for the error. Sound reasonable, I will fix this.